### PR TITLE
Update to coverlet 2.6.0

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: test/Directory.Build.props [hunk 1]
```diff
@@ -35,6 +35,7 @@
 
   <PropertyGroup Condition="'$(Coverage)' == 'true'">
       <CollectCoverage>true</CollectCoverage>
+      <SingleHit>true</SingleHit>
```
Comment (TomFinley) [inline test/Directory.Build.props]: This I guess is the major change, where you just register whether a line is hit or not, vs. count of hits, etc.?

**Comment by TomFinley** [comment]
If I pick a code coverage run arbitrarily, I see #2808 finished in 38m 13s. Yours finished in 23m 36s. That's a positive perf difference!

**Comment by sharwell** [comment]
@TomFinley it's a substantial improvement in time, but some of the data seems to be missing. I'm still investigating.

File: build/Codecoverage.proj [hunk 2]
```diff
@@ -17,8 +17,8 @@
       <_BranchName Condition="'$(_BranchName)' == ''">$(BUILD_SOURCEBRANCHNAME)</_BranchName>
     </PropertyGroup>
 
-    <Message Importance="high" Text="&quot;$(_ReportGeneratorPath)&quot; -reports:$(BaseOutputPath)$(PlatformConfig)\coverage\*.coverage -targetdir:$(BaseOutputPath)$(PlatformConfig)\coverage -reporttypes:Cobertura" />
-    <Exec Command="&quot;$(_ReportGeneratorPath)&quot; -reports:$(BaseOutputPath)$(PlatformConfig)\coverage\*.coverage -targetdir:$(BaseOutputPath)$(PlatformConfig)\coverage -reporttypes:Cobertura" />
+    <Message Importance="high" Text="&quot;$(_ReportGeneratorPath)&quot; -reports:$(BaseOutputPath)$(PlatformConfig)\coverage\*.coverage -targetdir:$(BaseOutputPath)$(PlatformConfig)\coverage -filefilters:+https* -reporttypes:Cobertura" />
+    <Exec Command="&quot;$(_ReportGeneratorPath)&quot; -reports:$(BaseOutputPath)$(PlatformConfig)\coverage\*.coverage -targetdir:$(BaseOutputPath)$(PlatformConfig)\coverage -filefilters:+https* -reporttypes:Cobertura" />
```
Comment (sharwell) [inline build/Codecoverage.proj]: 📝 This is an attempt to address a change in the way numbers were appearing in the new reports. For some reason, the file submitted to codecov had two copies of each source file, line this:

```xml
<class name="Microsoft.ML.StaticPipe.LdaFitResult" filename="D:\a\1\s\src\Microsoft.ML.StaticPipe\LdaStaticExtensions.cs" line-rate="0" branch-rate="1" complexity="NaN">
  <methods>
    <method name=".ctor" signature="(Microsoft.ML.Transforms.Text.LatentDirichletAllocationTransformer/LdaSummary)" line-rate="0" branch-rate="1">
      <line number="24" hits="0" branch="false" />
      <line number="25" hits="0" branch="false" />
      <line number="26" hits="0" branch="false" />
      <line number="27" hits="0" branch="false" />
    </method>
  </methods>
  <lines>
    <line number="24" hits="0" branch="false" />
    <line number="25" hits="0" branch="false" />
    <line number="26" hits="0" branch="false" />
    <line number="27" hits="0" branch="false" />
  </lines>
</class>
<class name="Microsoft.ML.StaticPipe.LdaFitResult" filename="https://raw.githubusercontent.com/dotnet/machinelearning/d495bb4c7a58c459e753b9de3d0895c188468b58/src/Microsoft.ML.StaticPipe/LdaStaticExtensions.cs" line-rate="1" branch-rate="1" complexity="NaN">
  <methods>
    <method name=".ctor" signature="(Microsoft.ML.Transforms.Text.LatentDirichletAllocationTransformer/LdaSummary)" line-rate="1" branch-rate="1">
      <line number="24" hits="1" branch="false" />
      <line number="25" hits="1" branch="false" />
      <line number="26" hits="1" branch="false" />
      <line number="27" hits="1" branch="false" />
    </method>
  </methods>
  <lines>
    <line number="24" hits="1" branch="false" />
    <line number="25" hits="1" branch="false" />
    <line number="26" hits="1" branch="false" />
    <line number="27" hits="1" branch="false" />
  </lines>
</class>
```

**Comment by sharwell** [comment]
This should be ready for review. The changes to coverage were primarily the following:

1. Microsoft.ML.Benchmarks and Microsoft.ML.Benchmarks.Tests are no longer reported. These files had no coverage previously.
2. Microsoft.ML.FSharp.Tests is no longer reported; the reportgenerator file filter might need to be updated to include it

The other coverage changes appear to be normal non-determinism in the coverage runs.

**Comment by sharwell** [comment]
For some reason the button to move this from a draft pull request to a normal pull request is missing here...

**Comment by TomFinley** [comment]
> For some reason the button to move this from a draft pull request to a normal pull request is missing here...

Interesting. Maybe it's one of those things that somehow needs write access? But that doesn't make sense. Anyway, I pushed the button...

File: build/ci/phase-template.yml [hunk 3]
```diff
@@ -40,9 +40,6 @@ phases:
     - ${{ if eq(parameters.name, 'MacOS') }}:
       - script: brew update && brew install libomp mono-libgdiplus gettext && brew link gettext --force
         displayName: Install runtime dependencies
-    - powershell: $(Build.SourcesDirectory)/build/PatchVSTestConsole.ps1
-      displayName: Hot patch vstest.console
```
Comment (sharwell) [inline build/ci/phase-template.yml]: 📝 I'm testing without this change, _hoping_ that the cause of coverage problems with 2.6.0 was not related to this specific change.

Comment (sharwell) [reply build/ci/phase-template.yml]: Removed this change. We can bring it back if problems start occurring.

File: test/Directory.Build.props (hunk 1 above)
Comment (codemzs) [reply test/Directory.Build.props]: yep, he made this change in the coverlet tool! Thanks, Sam!

---
In reply to: [262544802](https://github.com/dotnet/machinelearning/pull/2843#discussion_r262544802) [](ancestors = 262544802)

**Comment by codemzs** [comment]
Thanks, @sharwell !

**Pull request merged**